### ISSUE TITLE
Travis-CI

### ISSUE THREAD

**Issue by davidhjones** [codethesouth/website#9]
I've added this repo to Travis-CI. Need to work with Tyler/Grant to hook everything up

This issue should include integration as well as updating the README to give a glance into the CI status.

**Comment by codewithtyler** [comment]
Awesome, so for reference anyone wanting to see the most recent build done using Travis CI you can [find it here](https://travis-ci.org/codethesouth/website). At the moment we currently don't have any active or past builds but that will change as things progress.

I'd also like to note that we can add the following markdown to our README and it will give anyone viewing the repo an immediate glance at whether or not our CI build is passing or failing.

``` markdown
[![Build Status](https://travis-ci.org/codethesouth/website.svg?branch=master)](https://travis-ci.org/codethesouth/website)
```

Last on my list of things I wanted to mention is that you can [find the documentation here](https://docs.travis-ci.com/user/deployment/custom/) for custom deployment via FTP. You'll notice that it has you put the FTP username and password in the `.travis.yml`. Actually we can do is in that file we can have those lines point to a couple of environment variables. The environment variables will be added via the Travis CI settings for the repo. This way the FTP username and password is not visible to the public. I'd also like to note that any collaborators on the GitHub project (not sure about those who just have push rights and nothing else) have access to the environment variables in Travis CI. So we will need to make sure that only those who need access get it.

That should get us started on the deployment part of using Travis CI. We can talk about concatenating and minifying files later.

**Comment by granteous** [comment]
Ok I have to get working on the FTP on simple helix which means I need to dedicate a user to the production environment. We don't need root running the processes. This same user will also be the FTP credentials.

**Comment by codewithtyler** [comment]
Sounds good. I'll see if I can find some time tonight to setup Grunt with combining and minifying those CSS and JSU files. Once I get that merged in and you've got the FTP user created I'll take a look at setting up Travis for the deployment.

@davidhjones can you create another branch called release or prod or something?

**Comment by davidhjones** [comment]
Yeah I'll create the two branches tonight - one for development, and one for production. Production will be setup for Travis CI

**Comment by granteous** [comment]
Take a look at the trello board before-hand. I created a Branches and Versioning card for us to come up with the branching and versioning structure. I gave an example of one way we could do it but I’d like to hear from Tyler so we can nail something down. Basically before we create a development branch and a production branch we need to come up with version numbers for each. This may not be important for the website repo because we won’t have multiple CodeTheSouth websites out in the wild but as far as the TrolleyTracking app I think we need it. My thought here is that in the future there will be multiple TrolleyTracking applications in use and not every “customer” will be on the same version of the app and the process of upgrading versions could be a lengthy process in the government systems.
